### PR TITLE
ENH: add sami2py model object

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #44 

Adds an instrument object for sami2py, including download support for the test data file hosted at the sami2py github page. 

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

```
import datetime as dt
import pysat
from pysatModels.models import sami2py_sami2
sami2 = pysat.Instrument(inst_module=sami2py_sami2, tag='test')
sami2.download(dt.datetime(2019, 1, 1), dt.datetime(2019, 1, 1))
sami2.load(2019, 1)
```

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
Putting this up for discussion at the next meeting.  The test failure is related to a bug in pysat xarray instruments that has been fixed in pysat/pysat#441.

**Comment by jklenzing** [comment]
TODO before merge-ready:
- [ ] Path forward on naming conventions (platform, name)
- [ ] Make sure appropriate warnings are everywhere
- [ ] reset travis.yml to use pysat develop-3, not icon_updates

File: pysatModels/models/sami2py_sami2.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    Parameters
+    ----------
+    self : pysat.Instrument
+        This object
+
+    """
+
+    print("".join(["References and information about samip2y are available at ",
+                   "https://sami2py.readthedocs.io/en/latest/introduction.html",
+                   "#references"]))
```
Comment (aburrell) [inline pysatModels/models/sami2py_sami2.py:72]: Shouldn't this be logging at the info level?

File: pysatModels/models/sami2py_sami2.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                                                           month=date.month,
+                                                           day=date.day))
+        remote_path = '/'.join((remote_url.strip('/'), remote_path.strip('/'),
+                                fname))
+        req = requests.get(remote_path)
+        if req.status_code != 404:
+            open(saved_local_fname, 'wb').write(req.content)
+
+    else:
+        warnings.warn('Not implemented in this version.')
```
Comment (aburrell) [inline pysatModels/models/sami2py_sami2.py:197]: Perhaps change warning to indicate this is only intended to be used for accessing the test files?

**Comment by aburrell** [comment]
Also, if changing to use a `setup.cfg` file, we should put the info from the `.coveragerc` file inside there and change the coverage/coveralls calls to refer to the `setup.cfg` file.

**Pull request merged**